fairlearn/datasets/_fetch_diabetes_hospital.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+import pathlib
+
+from sklearn.datasets import fetch_openml
+from ._constants import _DOWNLOAD_DIRECTORY_NAME
+
+
+def fetch_diabetes_hospital(
+    *, cache=True, data_home=None, as_frame=False, return_X_y=False
+):
+    """Load the preprocessed Diabetes 130-Hospitals dataset (binary classification).
+
+    Download it if necessary.
+
+    ==============   ============================
+    Samples total                          101766
+    Dimensionality                             25
+    Features         numeric, categorical, string
+    Classes                                     2
+    ==============   ============================
+
+    Source: UCI Repository :footcite:`strack2014diabetes`
+    Paper: Strack et al., 2014 :footcite:`strack2014impact`
+
+    The "Diabetes 130-Hospitals" dataset represents 10 years of clinical care at 130
+    U.S. hospitals and delivery networks, collected from 1999 to 2008. Each record
+    represents the hospital admission record for a patient diagnosed with diabetes
+    whose stay lasted between one to fourteen days.
+
+    The original "Diabetes 130-Hospitals" dataset was collected by Beata Strack,
+    Jonathan P. DeShazo, Chris Gennings, Juan L. Olmo, Sebastian Ventura,
+    Krzysztof J. Cios, and John N. Clore in 2014.
+
+    This version of the dataset was derived by the Fairlearn team for the SciPy
+    2021 tutorial "Fairness in AI Systems: From social context to practice using
+    Fairlearn". In this version, the target variable "readmitted" is binarized
+    into whether the patient was re-admitted within thirty days.
+
+    # TODO: Read more in the :ref:`User Guide <diabetes_hospital_data>`.
```

Comment:
TODO?

Reply:
TODO as in, I still needed to write the User Guide :)

Reply:
So should the PR be in "draft" mode or "WIP" (work in progress)?

Reply:
It should have been when I opened the PR, you are right. Now that the user guide has been done last week, it was ready for review.